src/metabase/models/api_key.clj

```diff
@@ -12,6 +16,31 @@
 
 (methodical/defmethod t2/table-name :model/ApiKey [_model] :api_key)
 
+(mi/define-batched-hydration-method add-group-name
+  :group_name
+  "Add to each ApiKey a single group_name. Assume that each ApiKey is a member of either zero or one groups other than
+  the 'All Users' group."
+  [api-keys]
+  (when (seq api-keys)
+    (let [api-key-id->permissions-groups
+          (group-by :api-key-id
+                    (t2/query {:select [[:pg.name :group-name]
+                                        [:pg.id :group-id]
+                                        [:api_key.id :api-key-id]]
+                               :from [[:permissions_group :pg]]
+                               :join [[:permissions_group_membership :pgm]
+                                      [:= :pgm.group_id :pg.id]
+                                      :api_key [:= :api_key.user_id :pgm.user_id]]
+                               :where [:in :api_key.id (map u/the-id (t2/select :model/ApiKey))]}))
```

Comment:
shouldn't this query be restricted to the IDs of `api-keys` passed in as an argument, instead of all API keys?

Reply:
Argh, yes, thank you - I was sending this to CIDER and forgot to change it back. Nice catch.